src/rpc/scgi_task.cc

```diff
@@ -239,36 +248,99 @@ SCgiTask::event_error() {
   close();
 }
 
+// Convenience function similar to zlib's compress(), but uses a gzip
+// header. Returns an empty string on error, as even a compressed
+// empty string will have data.
+std::string
+gzip_compress(const char* buffer, uint32_t length) {
```

Comment:
Move to `src/utils`.

Reply:
This was changed to be a member function, if it becomes generic again I'll split it out to src/utils/.